hilti/toolchain/src/compiler/codegen/expressions.cc

```diff
@@ -41,182 +48,208 @@ struct Visitor : hilti::visitor::PreOrder<cxx::Expression, Visitor> {
         cxx::Block block;
         cg->pushCxxBlock(&block);
         auto arguments =
-            util::join(node::transform(n.arguments(), [this](auto& x) { return cg->compile(x, lhs); }), ", ");
+            util::join(node::transform(n->arguments(), [this](auto& x) { return cg->compile(x, lhs); }), ", ");
         cg->popCxxBlock();
 
-        block.addStatement(fmt("%s(%s)", cxx::ID(n.cxxname()), arguments));
+        block.addStatement(fmt("%s(%s)", cxx::ID(n->cxxname()), arguments));
 
         cxx::Formatter f;
         f << block;
-        return fmt("(%s)", f.str());
+        result = fmt("(%s)", f.str());
     }
 
-    result_t operator()(const expression::Coerced& n) {
-        return cg->coerce(cg->compile(n.expression(), lhs), n.expression().type(), n.type());
+    void operator()(expression::Coerced* n) final {
+        result = cg->coerce(cg->compile(n->expression(), lhs), n->expression()->type(), n->type());
     }
 
-    result_t operator()(const expression::Ctor& n) { return cg->compile(n.ctor(), lhs); }
+    void operator()(expression::Ctor* n) final { result = cg->compile(n->ctor(), lhs); }
 
-    result_t operator()(const expression::Deferred& n) {
-        auto type = cg->compile(n.type(), codegen::TypeUsage::Storage);
-        auto value = cg->compile(n.expression());
+    void operator()(expression::Deferred* n) final {
+        auto type = cg->compile(n->type(), codegen::TypeUsage::Storage);
+        auto value = cg->compile(n->expression());
 
-        if ( n.catchException() )
-            return fmt(
-                "::hilti::rt::DeferredExpression<%s>([=]() -> %s { try { return %s; } catch ( ... ) { return "
-                "::hilti::rt::result::Error(\"n/a\"); } })",
-                type, type, value);
+        if ( n->catchException() )
+            result =
+                fmt("::hilti::rt::DeferredExpression<%s>([=]() -> %s { try { return %s; } catch ( ... ) { return "
+                    "{}; } })",
```

Comment:
Does this now swallow errors? If yes, could you add a comment documenting the semantics?